crates/common/src/abi_codec.rs

```diff
@@ -521,21 +527,33 @@ mod tests {
         }
 
         #[test]
-        fn test_arrays_not_supported() {
-            let slot = AbiSlot {
-                name: "arr".into(),
-                ty: AbiType::Array {
+        fn test_invalid_fixed_size_array_decode() {
+            // Two u32 elements require 4 M31 slots; provide only 3 -> insufficient data
+            let src = vec![M31::from(1u32), M31::from(2u32), M31::from(3u32)];
+            let result = decode_one(
+                &AbiType::FixedSizeArray {
                     element: Box::new(AbiType::U32),
-                    size: Some(3),
+                    size: 2,
                 },
-            };
-            let input = InputValue::List(vec![1.into(), 2.into(), 3.into()]);
-            let result = encode_input_args(&[slot], &[input]);
-            assert!(result.is_err());
-            assert!(result
-                .unwrap_err()
-                .to_string()
-                .contains("Array types are not yet supported"));
+                &src,
+                0,
+            );
+            assert!(matches!(result, Err(AbiCodecError::InsufficientData)));
+        }
+
+        #[test]
+        fn test_invalid_fixed_size_array_encode_size_mismatch() {
+            // Encoding should validate array length against the declared fixed size
+            let params = [AbiSlot {
+                name: "arr".into(),
+                ty: AbiType::FixedSizeArray {
+                    element: Box::new(AbiType::Felt),
+                    size: 3,
+                },
+            }];
+            let values = [InputValue::List(vec![0.into(), 1.into()])]; // only 2 provided
+            let err = encode_input_args(&params, &values).unwrap_err().to_string();
+            assert!(err.contains("array size mismatch"));
```

Comment:
Should the assert ensure the error is a `AbiCodecError::TypeMismatch` and not just assert the error string?